pkg/persistence/object_store_test.go

```diff
@@ -62,6 +62,99 @@ func newObjectBackupStoreTestHarness(bucket, prefix string) *objectBackupStoreTe
 	}
 }
 
+func TestIsValid(t *testing.T) {
+	tests := []struct {
+		name        string
+		prefix      string
+		storageData cloudprovider.BucketData
+		expectErr   bool
+	}{
+		{
+			name:      "empty backup store with no prefix is valid",
+			expectErr: false,
+		},
+		{
+			name:      "empty backup store with a prefix is valid",
+			prefix:    "bar",
+			expectErr: false,
+		},
+		{
+			name: "backup store with no prefix and only unsupported directories is invalid",
+			storageData: map[string][]byte{
+				"backup-1/ark-backup.json": {},
+				"backup-2/ark-backup.json": {},
+			},
+			expectErr: true,
+		},
+		{
+			name:   "backup store with a prefix and only unsupported directories is invalid",
+			prefix: "backups",
+			storageData: map[string][]byte{
+				"backups/backup-1/ark-backup.json": {},
+				"backups/backup-2/ark-backup.json": {},
+			},
+			expectErr: true,
+		},
+		{
+			name: "backup store with no prefix and both supported and unsupported directories is invalid",
```

Comment:
So the logic here is to make sure Ark still owns the root of the path it's given, whether that's the bucket root or a prefix inside it.

Is it this just to avoid future name collisions? Additionally, will the directory specified for restic be considered invalid in the current code, since it's only 2 hard-coded values?

Reply:
Yeah, this is to ensure Ark owns the root location, and also that the migration was done properly.

It will, but I'm working on another PR for restic to share the bucket, at which point it'll be a valid dir.